server/public/shared/mlog/tlog.go

```diff
@@ -51,8 +58,48 @@ func CreateConsoleTestLogger(useJSON bool, level Level) *Logger {
 
 	target := targets.NewWriterTarget(os.Stdout)
 	if err := logger.log.Logr().AddTarget(target, "_testcon", filter, formatter, 1000); err != nil {
-		panic(err)
+		tb.Fatalf("failed to add target %v", err)
+	}
+
+	tb.Cleanup(func() {
+		err := logger.Shutdown()
+		if err != nil {
+			tb.Fatalf("failed to shut down test logger %v", err)
+		}
+	})
+
+	return logger
+}
+
+// CreateTestLogger creates a logger for unit tests. Log records are output via `t.Log`.
+// All log messanges with level trace or lower are logged.
+// The returned logger get Shutdown() when the tests completes. The caller should not shut it down.
+func CreateTestLogger(t *testing.T) *Logger {
```

Comment:
This looks a lot like the test target and logger here:  https://github.com/mattermost/logr/blob/918322db6b5d815477999cac8060a655fb1ab9ac/targets/testing.go#L53

Completely fine with a customer one in mlog, just wanted you to be aware.

Reply:
Given that `mlog` uses it's own `Logger` wrapper I found it easier to create the `Logger` using `NewLogger` instead of `targets.CreateTestLogger` and wrapping it.